src/condor_shadow.V6.1/pseudo_ops.cpp

```diff
@@ -1049,6 +1049,38 @@ pseudo_event_notification( const ClassAd & ad ) {
 }
 
 
+bool
+LookupIntInContext( const ClassAd & ad, const std::string & attr, int & value ) {
+	auto * ctx = ad.Lookup( ATTR_CONTEXT_AD );
+	const ClassAd * context = dynamic_cast<ClassAd *>(ctx);
+	if( context ) {
+		return context->LookupInteger( attr, value );
+	}
+	return false;
+}
```

Comment:
We shouldn't fix this here, but if classads had a templatized 
```
template<typename T> bool Lookup(const ClassAd &ad, const std::string &name, T &result);
```

A lot of this code could be condensed.

Reply:
Agreed.